### PR TITLE
[CI:DOCS] Document how to get secret mounts working on RHEL8

### PR DESCRIPTION
**Pull request by imphil**
Add a section to the troubleshooting guide describing how to get secret mounts working in RHEL8, as discussed in
https://github.com/containers/buildah/issues/4228.

It took me a while to find this information, and a RHEL8 default setup isn't that uncommon, so I thought it would be worth the change.

### CODE REVIEW

File: troubleshooting.md [hunk 1]
```diff
@@ -1399,3 +1399,25 @@ first process can acquire it, this type of `image not known` error can arise.
 The maintainers of Podman have considered heavier-duty locks to close this
 timing window. However, the slowdown that all Podman commands would encounter
 was not considered worth the cost of completely closing this small timing window.
+
+### 41) A podman build step with `--mount=type=secret` fails with "operation not permitted"
+
+Executing a step in a `Dockerfile`/`Containerfile` which mounts secrets using `--mount=type=secret` fails with "operation not permitted" when running on a host filesystem mounted with `nosuid` and when using the `runc` runtime (both of which are true by default on RHEL8).
```
Comment (rhatdan) [inline troubleshooting.md:1405]: Take out comments on RHEL8, they will not age well.

File: troubleshooting.md [hunk 2]
```diff
@@ -1399,3 +1399,25 @@ first process can acquire it, this type of `image not known` error can arise.
...
+
+```
+STEP 3/13: RUN --mount=type=secret,id=MY_USER,target=/etc/dnf/vars/MY_USER     --mount=type=secret,id=MY_USER,target=/etc/dnf/vars/MY_USER     ...: time="2023-06-13T18:04:59+02:00" level=error msg="runc create failed: unable to start container process: error during container init: error mounting \"/var/tmp/buildah2251989386/mnt/buildah-bind-target-11\" to rootfs at \"/etc/dnf/vars/MY_USER\": mount /var/tmp/buildah2251989386/mnt/buildah-bind-target-11:/etc/dnf/vars/MY_USER (via /proc/self/fd/7), flags: 0x1021: operation not permitted"
+: exit status 1
+ERRO[0002] did not get container create message from subprocess: EOF
+```
+
+#### Solution
+
+* Install `crun`, e.g. with `dnf install crun` on RHEL8.
```
Comment (rhatdan) [inline troubleshooting.md:1420]: Remove `on RHEL8`

File: troubleshooting.md [hunk 3]
```diff
@@ -1399,3 +1399,25 @@ first process can acquire it, this type of `image not known` error can arise.
...
+* Use the `crun` runtime by passing `--runtime /usr/bin/crun` to `podman buildx build` (or `podman build` with `DOCKER_BUILDKIT=1` set).
```
Comment (rhatdan) [inline troubleshooting.md:1421]: Podman build is the same  podman buildx build.  Also DOCKER_BUILDKIT=1 is ignored.
podman build always does the buildkit mode.

**Comment by imphil** [comment]
Thanks for your review, @rhatdan. Updated.

**Comment by rhatdan** [comment]
/approve
LGTM
@giuseppe PTAL

File: troubleshooting.md [hunk 4]
```diff
@@ -1399,3 +1399,25 @@ first process can acquire it, this type of `image not known` error can arise.
...
 timing window. However, the slowdown that all Podman commands would encounter
 was not considered worth the cost of completely closing this small timing window.
+
+### 41) A podman build step with `--mount=type=secret` fails with "operation not permitted"
+
+Executing a step in a `Dockerfile`/`Containerfile` which mounts secrets using `--mount=type=secret` fails with "operation not permitted" when running on a host filesystem mounted with `nosuid` and when using the `runc` runtime.
+
+#### Symptom
+
+A `RUN` line in `Dockerfile`/`Containerfile` contains a [secret mount](https://github.com/containers/common/blob/main/docs/Containerfile.5.md) such as `--mount=type=secret,id=MY_USER,target=/etc/dnf/vars/MY_USER`.
```
Comment (TomSweeneyRedHat) [inline troubleshooting.md:1409]:
```suggestion
A `RUN` line in the `Dockerfile`/`Containerfile` contains a [secret mount](https://github.com/containers/common/blob/main/docs/Containerfile.5.md) such as `--mount=type=secret,id=MY_USER,target=/etc/dnf/vars/MY_USER`.
```

File: troubleshooting.md [hunk 5]
```diff
@@ -1399,3 +1399,25 @@ first process can acquire it, this type of `image not known` error can arise.
...
+When running `podman buildx build` the process fails with an error message like
```
Comment (TomSweeneyRedHat) [inline troubleshooting.md:1410]: soft suggestion
```suggestion
When running `podman buildx build` the process fails with an error message like:
```

Comment (rhatdan) [reply troubleshooting.md:1410]: Remove reference to buildx.  Should just be podman build

**Comment by imphil** [comment]
Thanks @TomSweeneyRedHat and @rhatdan, updated with your suggestions.

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
TYVM @imphil !

**Pull request merged**